include/lsst/afw/geom/SpanSet.h

```diff
@@ -0,0 +1,252 @@
+    SpanSet transformedBy(AffineTransform const & t) const;
+    SpanSet transformedBy(XYTransform const & t) const;
+
+    bool overlaps(SpanSet const & other) const;
+
+    bool contains(SpanSet const & other) const;
+    bool contains(Point2I const & point) const;
+
+    Point2D computeCentroid() const;
+
+    ellipses::Quadrupole computeShape() const;
+
+    SpanSet dilate(int r, Stencil s = Stencil::CIRCLE) const;
+    SpanSet dilate(SpanSet const & other) const;
+
+    SpanSet erode(int r, Stencil s = Stencil::CIRCLE) const;
+    SpanSet erode(SpanSet const & other) const;
+
+    template <typename Pixel, int inC>
+    ndarray::Array<Pixel, 1, 1> flatten(ndarray::Array<Pixel, 2, inC>  & input,
+                                                       Point2I const xy0 = Point2I()) const {
+        // Populate a one dimensional array with the values from input at taken at the points of SpanSet
+        ndarray::Array<Pixel, 1, 1> outputArray = ndarray::allocate(ndarray::makeVector(getArea()));
+        outputArray.deep() = 0;
+        flatten(outputArray, input, xy0);
+        return outputArray;
+    }
+
+    template <typename Pixel, int outC, int inC>
+    void flatten(
+            ndarray::Array<Pixel, 1, outC> & output,
+            ndarray::Array<Pixel, 2, inC> & input,
+            Point2I const xy0 =  Point2I()) const {
+        // Populate array output with values from input at positions given by SpanSet
+        applyFunctor([](Point2I point, int n, typename details::FlatNdGetter<Pixel>::Reference out,
+                        typename details::ImageNdGetter<Pixel, 2, inC>::Reference in) {out = in;},
+                     ndarray::ndFlat(output), ndarray::ndImage(input, xy0));
+    }
+
+    template <typename Pixel, int inC>
+    ndarray::Array<Pixel, 2, 2> unflatten(ndarray::Array<Pixel, 1, inC> & input) const {
+        // Create a two dimensional array the size of the bounding box. Populate values from input, placed at
+        // locations corresponding to SpanSet, offset by the lower corner of the bounding box
+        ndarray::Array<Pixel, 2, 2> outputArray = ndarray::allocate(bBox.getHeight(), bBox.getWidth());
+        outputArray.deep() = 0;
+        unflatten(outputArray, input, Point2I(bBox.getMinX(), bBox.getMinY()));
+        return outputArray;
+    }
+
+    template <typename Pixel, int outC, int inC>
+    void unflatten(ndarray::Array<Pixel, 2, outC> & output,
+                                  ndarray::Array<Pixel, 1, inC> & input,
+                                  Point2I const xy0 = Point2I()) const {
+        // Populate 2D ndarray output with values from input, at locations defined by SpanSet, optionally
+        // offset by xy0
+        applyFunctor([](Point2I point, int n, typename details::ImageNdGetter<Pixel, 2, outC>::Reference out,
+                        typename details::FlatNdGetter<Pixel>::Reference in) {out = in;},
+                        ndarray::ndImage(output, xy0), ndarray::ndFlat(input));
```

Comment:
Could you use more whitespace in multi-line lambdas like this one (there are a few more in this file) to make them easier to read?  At the very least, the lambda implementation should not be on the same line as part of the argument list if the argument list consumes more than one line.  For example, I'd reformat this as:

```
applyFunctor(
    [](
        Point2I point,
        int n,
        typename details::ImageNdGetter<Pixel, 2, outC>::Reference out,
        typename details::FlatNdGetter<Pixel>::Reference in
    ) {
        out = in;
    },
    ndarray::ndImage(output, xy0),
    ndarray::ndFlat(input)
);
```

(Also, you should pass the `Point` by const reference here as well).